File: Lectures/05-Collections-of-Objects/customer.cc
#include <vector>
#include <iostream>
using namespace std;
#include "customer.hh"
#include "ticket.hh"

void Customer::addTicket(Ticket* theTicket) {
  myTickets.push_back(theTicket);
}

int Customer::countTickets(void) {
  return myTickets.size();
}

vector<Ticket*> Customer::getTickets() {
  return myTickets;
}

void Customer::describeTicket(int theTicketNumber) {
  if (0 <= theTicketNumber && myTickets.size() > theTicketNumber) {
    cout << myTickets[theTicketNumber]->toString() << endl;
  }
}

File: Lectures/05-Collections-of-Objects/Customer.java
import java.util.ArrayList;

public class Customer {
  private ArrayList<Ticket> myTickets; 

  public Customer() {
    myTickets = new ArrayList<>(); 
  }

  public void addTicket(Ticket theTicket) {
    myTickets.add(theTicket); // Note that we no longer need a dedicated variable for theTicket
  }

  public int countTickets() {
    return myTickets.size();
  }

  public ArrayList<Ticket> getTickets() { // CAREFUL HERE
    return myTickets;                     // This is *not* a good idea to return
  }                                       // your internal and private attributes

  public void describeTicket(int ticketNumber) {
    if (0 <= ticketNumber   // We start at index 0
        && myTickets.size() > ticketNumber) { // size() is just outside of the collection.
      String details = myTickets.get(ticketNumber).toString();
      System.out.println(details);
    }
  }
}
